uips/uip-4.md

```diff
@@ -0,0 +1,102 @@
+
+### Encryption of Spend Backreference
+
+The `backref_commitment` should be encrypted using `ChaCha20-Poly1305`. [RFC-8349](https://datatracker.ietf.org/doc/rfc8439/) specifies that (key, nonce) pairs MUST NOT be reused.
+
+We derive a symmetric key $k$ from the `OutgoingViewingKey` $ovk$ using the BLAKE2b-512 hash function and personalization string `"Penumbra_Backref"`:
+
+```
+k = BLAKE2b-512("Penumbra_Backref", ovk)
+```
+
+One advantage of using a single key is that we can scan all spends using this key without having to do key derivation each time.
+
+A random 16-byte nonce $n$ will be generated and provided as a prefix in the `backref_commitment` field.
+
+Encryption is performed using the $(k, n)$ tuple and outputs the 32-byte ciphertext of the 32-byte note commitment, and a 16-byte MAC. The transmitted data in the `backref_commitment` field consists of a concatenation of the nonce, ciphertext and MAC:
+
+```
+backref_commitment = n || ciphertext || MAC
+```
+
+The `backref_commitment` is thus 64 bytes (16 byte nonce + 32 byte ciphertext + 16 byte MAC).
+
+### EffectHash
+
+Currently the `EffectHash` for the `Spend` action is computed as:
+
+`effect_hash = BLAKE2b-512(len(type_url) || type_url || proto_encode(proto))`
+
+where `type_url` is the bytes of a variable-length Type URL defining the proto message, `len(type_url)` is the length of the Type URL encoded as 8 bytes in little-endian  order, and `proto` represents the proto used to represent the effecting data, and `proto_encode` represents encoding the proto message as a vector of bytes.
+
+#### Backwards Compatibility
+
+The `EffectHash` computation is unchanged if the new `backref_commitment` field is not populated. This is the behavior provided by making the field `optional` in the `SpendBody` protocol message. However, if the `backref_commitment` field is populated, then it is included in the `EffectHash` computation per the existing `proto_encode` method as described above.
+
+### Transaction Perspectives and Views
+
+The `TransactionPerspective` and `TransactionView` will be unchanged. The backreference is treated as an internal sync optimization detail.
+
+## Rationale
+
+ZCash has considered a similar approach wherein backwards syncing is enabled using references in the memo fields. The advantage of this approach is that DAGSync-aware clients can populate these fields without a change to the consensus rules.
+
+## Backwards Compatibility
+
+Since the `backref_commitment` field is proposed to be optional, there should be no compatibility issues. The `EffectHash` for a `Spend` will be unchanged if the `backref_commitment` field is absent. Once all clients have added `backref_commitment` support, a future UIP could make the field mandatory.
```

Comment:
In a Privacy Considerations section (see https://github.com/penumbra-zone/UIPs/pull/3), I think it would be good to discuss:

- The presence or absence of the `encrypted_backref` field as being a distinguisher for client software, and how that tradeoff was considered against compatibility issues to reach this conclusion